### ISSUE TITLE
Linux Desktop app on start up does not proceed to login (DeprecationWarning)

### ISSUE THREAD

**Issue by AJPoulter** [bitwarden/clients#6910]
### Steps To Reproduce

1. Open terminal
2. Type `bitwarden-desktop` and press `enter`. Application launches showing animated image
3. View warning details written to `stdout` in terminal

### Expected Result

Bitwarden desktop app starts and user can login and access vault entries

### Actual Result

Bitwarden desktop app starts but only displays rotating animated "getting ready" image and does not display window where user can enter password to open vault.
Terminal displays the following:

` ~> bitwarden-desktop 
(node:5108) DeprecationWarning: file property is deprecated and will be removed in v5.
(Use `electron --trace-deprecation ...` to show where the warning was created)
[5108:1118/084840.874328:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[5108:1118/084840.874384:ERROR:select_file_dialog_linux_portal.cc(274)] Failed to read portal version property
(node:5108) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'init')
    at BiometricsService.<anonymous> (/usr/lib/bitwarden/app.asar/main.js:47305:55)
    at Generator.next (<anonymous>)
    at /usr/lib/bitwarden/app.asar/main.js:47271:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/usr/lib/bitwarden/app.asar/main.js:47267:12)
    at BiometricsService.init (/usr/lib/bitwarden/app.asar/main.js:47304:16)
    at Main.<anonymous> (/usr/lib/bitwarden/app.asar/main.js:54050:46)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/bitwarden/app.asar/main.js:53942:58)
(node:5108) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
[5108:1118/084841.068867:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop
08:48:41.760 › State version: 8
08:48:41.767 › Migrator ba (to version 2) should migrate: false - up
08:48:41.769 › Migrator la (to version 3) should migrate: false - up
08:48:41.770 › Migrator ua (to version 4) should migrate: false - up
08:48:41.770 › Migrator ha (to version 5) should migrate: false - up
08:48:41.771 › Migrator ga (to version 6) should migrate: false - up
08:48:41.771 › Migrator va (to version 7) should migrate: false - up
08:48:41.772 › Migrator ma (to version 8) should migrate: false - up
08:48:41.893 › Error: The name org.freedesktop.secrets was not provided by any .service files
08:48:41.896 › Error: The name org.freedesktop.secrets was not provided by any .service files
08:48:41.921 › Error: The name org.freedesktop.secrets was not provided by any .service files
(node:5108) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of null (reading 'null')
    at new Menubar (/usr/lib/bitwarden/app.asar/main.js:45647:131)
    at MenuMain.<anonymous> (/usr/lib/bitwarden/app.asar/main.js:45701:56)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/bitwarden/app.asar/main.js:45671:58)
(node:5108) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
08:48:42.137 › Unable to fetch ServerConfig: undefined
08:48:42.195 › Error: The name org.freedesktop.secrets was not provided by any .service files
08:48:42.345 › Unable to fetch ServerConfig: undefined
(node:5108) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of null (reading 'null')
    at new Menubar (/usr/lib/bitwarden/app.asar/main.js:45647:131)
    at MenuMain.<anonymous> (/usr/lib/bitwarden/app.asar/main.js:45701:56)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/bitwarden/app.asar/main.js:45671:58)
(node:5108) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
08:48:44.845 › Error: The name org.freedesktop.secrets was not provided by any .service files
(node:5108) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of null (reading 'null')
    at new Menubar (/usr/lib/bitwarden/app.asar/main.js:45647:131)
    at MenuMain.<anonymous> (/usr/lib/bitwarden/app.asar/main.js:45701:56)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/bitwarden/app.asar/main.js:45671:58)
(node:5108) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 5)
08:53:41.372 › Error: The name org.freedesktop.secrets was not provided by any .service files
08:53:41.530 › Unable to fetch ServerConfig: undefined`

### Operating System

Linux

### Operating System Version

EndeavourOs Archlinux x86_64, Kernel 6.6.1-zen1-1-zen

### Installation method

Other

### Build Version

Version 2023.10.1 Shell 25.9.4 Renderer 114.0.5735.289 Node 18.15.0 Architecture x64

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by oktupol** [comment]
I'm getting a similar behaviour. The window pops up for a split second before the application crashes:

```
(node:2) DeprecationWarning: file property is deprecated and will be removed in v5.
(Use `bitwarden --trace-deprecation ...` to show where the warning was created)
(node:2) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'init')
    at BiometricsService.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47305:55)
    at Generator.next (<anonymous>)
    at /app/Bitwarden/resources/app.asar/main.js:47271:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/app/Bitwarden/resources/app.asar/main.js:47267:12)
    at BiometricsService.init (/app/Bitwarden/resources/app.asar/main.js:47304:16)
    at Main.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:54050:46)
    at Generator.next (<anonymous>)
    at fulfilled (/app/Bitwarden/resources/app.asar/main.js:53942:58)
(node:2) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

**Comment by Adedamola-Aina** [comment]
Hi there,

I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below.

Thanks!

**Comment by user426** [comment]
Same here on flatpak desktop 

############ERROR LOG###############
`✦ ❯ flatpak run com.bitwarden.desktop
(node:2) DeprecationWarning: file property is deprecated and will be removed in v5.
(Use `bitwarden --trace-deprecation ...` to show where the warning was created)
(node:2) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'init')
    at BiometricsService.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47305:55)
    at Generator.next (<anonymous>)
    at /app/Bitwarden/resources/app.asar/main.js:47271:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/app/Bitwarden/resources/app.asar/main.js:47267:12)
    at BiometricsService.init (/app/Bitwarden/resources/app.asar/main.js:47304:16)
    at Main.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:54050:46)
    at Generator.next (<anonymous>)
    at fulfilled (/app/Bitwarden/resources/app.asar/main.js:53942:58)
(node:2) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
14:03:35.981 › Error: No password found
14:03:36.020 › Error: No password found
14:03:36.128 › Error: No password found
14:03:36.129 › Error: No password found
14:03:36.131 › Error: No password found
14:03:36.150 › Error: No password found
14:03:36.203 › TypeError: Cannot read properties of undefined (reading 'osSupportsBiometric')
    at BiometricsService.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47310:55)
    at Generator.next (<anonymous>)
    at /app/Bitwarden/resources/app.asar/main.js:47271:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/app/Bitwarden/resources/app.asar/main.js:47267:12)
    at BiometricsService.osSupportsBiometric (/app/Bitwarden/resources/app.asar/main.js:47309:16)
    at BiometricsService.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47320:58)
    at Generator.next (<anonymous>)
    at fulfilled (/app/Bitwarden/resources/app.asar/main.js:47268:58)
14:03:36.205 › TypeError: Cannot read properties of undefined (reading 'osSupportsBiometric')
    at BiometricsService.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47310:55)
    at Generator.next (<anonymous>)
    at /app/Bitwarden/resources/app.asar/main.js:47271:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/app/Bitwarden/resources/app.asar/main.js:47267:12)
    at BiometricsService.osSupportsBiometric (/app/Bitwarden/resources/app.asar/main.js:47309:16)
    at DesktopCredentialStorageListener.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47518:59)
    at Generator.next (<anonymous>)
    at /app/Bitwarden/resources/app.asar/main.js:47446:71
    at new Promise (<anonymous>)
14:03:36.318 › Error: No password found
14:03:36.320 › Error: No password found
14:03:36.456 › Error: No password found
14:03:37.003 › Error: No password found
14:03:37.102 › Error: No password found
`
####################################

**Comment by Supermath101** [comment]
> I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below.

This happens on my system when running the Sway Wayland Compositor (version 1.8.1), but not on the Cinnamon Desktop environment. However, I still run into the issue #6818. The crash after a split-second might be due to the tiling nature of Sway, or more likely a Wayland-exclusive issue. I might try this on I3 (a drop in replacement for Sway, based on X11) sometime later.

```
$ flatpak run com.bitwarden.desktop
(node:2) DeprecationWarning: file property is deprecated and will be removed in v5.
(Use `bitwarden --trace-deprecation ...` to show where the warning was created)
Gtk-Message: 12:55:47.116: Failed to load module "xapp-gtk3-module"
(node:2) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'init')
    at BiometricsService.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:47305:55)
    at Generator.next (<anonymous>)
    at /app/Bitwarden/resources/app.asar/main.js:47271:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/app/Bitwarden/resources/app.asar/main.js:47267:12)
    at BiometricsService.init (/app/Bitwarden/resources/app.asar/main.js:47304:16)
    at Main.<anonymous> (/app/Bitwarden/resources/app.asar/main.js:54050:46)
    at Generator.next (<anonymous>)
    at fulfilled (/app/Bitwarden/resources/app.asar/main.js:53942:58)
(node:2) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

Output of `uname -a`: `Linux fedora 6.5.10-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Nov  2 19:59:55 UTC 2023 x86_64 GNU/Linux`

```
$ flatpak info com.bitwarden.desktop

Bitwarden - A secure and free password manager for all of your devices

          ID: com.bitwarden.desktop
         Ref: app/com.bitwarden.desktop/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 2023.10.1
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 344.0 MB
     Runtime: org.freedesktop.Platform/x86_64/23.08
         Sdk: org.freedesktop.Sdk/x86_64/23.08

      Commit: ccdbf9a3897195c29079e848859e6a7eb9285499a7aec0960895802f019551cb
      Parent: d432973b779ec3664db1b0498842eb11bec8bbb4e7edc939c9c388540b565af3
     Subject: Enable wayland by default, fallback to X11 (#99) (1eef6b23)
        Date: 2023-11-12 10:11:25 +0000
```

**Comment by oktupol** [comment]
This issue has magically gone away for me. I'm still on the same Bitwarden version. I suspect some system update having had some impact. Perhaps even Sway. I am also using it, and there has been an update quite recently.

**Comment by timblaktu** [comment]
Same issue here on bleeding edge updated `6.6.3-arch1-1` install of `bitwarden 2023.10.1-1`. I'm running hyprland with no display manager (auto-logging-in via custom systemd service), and am initially assuming this is related to bitwarden needing/wanting a secrets vault, which I don't have now...

.. a quick googling found this issue all over the place for people with no secret keyring installed. In most cases it looks like installing something `gnome-keyring` or `kde-wallet` fixes it.

In my case, I don't want gnome* or kde* on my system, so I found the minimal solution:

**Comment by djsmith85** [comment]
@AJPoulter You mentioned [here](https://github.com/bitwarden/clients/issues/6945#issuecomment-1850007510) that #6945 is a duplicate of this issue. As #6945 has been resolved with `2024.2`, can this also be closed?

@oktupol, @user426, @Supermath101, @timblaktu: Could you please also check if `2024.2` resolves the issue?

**Comment by Supermath101** [comment]
> @AJPoulter You mentioned [here](https://github.com/bitwarden/clients/issues/6945#issuecomment-1850007510) that #6945 is a duplicate of this issue. As #6945 has been resolved with `2024.2`, can this also be closed?
> 
> @oktupol, @user426, @Supermath101, @timblaktu: Could you please also check if `2024.2` resolves the issue?

On Sway (Wayland compositor), it seems to work fine now. Thanks.
Output of `flatpak info com.bitwarden.desktop`:
```

Bitwarden - A secure and free password manager for all of your devices

          ID: com.bitwarden.desktop
         Ref: app/com.bitwarden.desktop/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 2024.2.0
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 357.4 MB
     Runtime: org.freedesktop.Platform/x86_64/23.08
         Sdk: org.freedesktop.Sdk/x86_64/23.08

      Commit: 005bc689d6ec515480b45a655e46a829a6708d91bb55ee2b1c0bd6cd3a029481
      Parent: ee5bd5c3c0fcc528d3daae12a25e0fb38acdfabfa0e43b99a2bb1a01ea050ff8
     Subject: Update 2 modules (7f83248d)
        Date: 2024-02-08 03:27:26 +0000
```
